src/test/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesTaskLauncherIntegrationIT.java

```diff
@@ -16,235 +16,122 @@
+		awaitWithPollAndTimeout(undeploymentTimeout())
+				.untilAsserted(() -> assertThat(taskLauncher().status(launchId).getState()).isEqualTo(LaunchState.unknown));
 	}
 
 	@Test
-	public void testTaskAdditionalContainer() {
-		log.info("Testing {}...", "TaskAdditionalContainer");
-
-		KubernetesTaskLauncher kubernetesTaskLauncher = new KubernetesTaskLauncher(new KubernetesDeployerProperties(),
-				new KubernetesTaskLauncherProperties(), kubernetesClient);
-
+	void cleanupDeletesTaskPod(TestInfo testInfo) {
+		logTestInfo(testInfo);
 		AppDefinition definition = new AppDefinition(randomName(), null);
 		Resource resource = testApplication();
-		Map<String, String> props = Collections.singletonMap("spring.cloud.deployer.kubernetes.additionalContainers",
-				"[{name: 'test', image: 'busybox:latest', command: ['sh', '-c', 'echo hello']}]");
-		AppDeploymentRequest request = new AppDeploymentRequest(definition, resource, props);
-
-		log.info("Launching {}...", request.getDefinition().getName());
-
-		String launchId = kubernetesTaskLauncher.launch(request);
-		Timeout timeout = deploymentTimeout();
-
-		await().pollInterval(Duration.ofMillis(timeout.pause))
-                .atMost(Duration.ofMillis(timeout.maxAttempts * timeout.pause))
-                .untilAsserted(() -> {
-			assertThat(taskLauncher().status(launchId).getState()).isEqualTo(LaunchState.running);
-        });
-
+		AppDeploymentRequest request = new AppDeploymentRequest(definition, resource, null);
 		String taskName = request.getDefinition().getName();
 
-		List<Pod> pods = kubernetesClient.pods().withLabel("task-name", taskName).list().getItems();
+		log.info("Launching {}...", taskName);
+		String launchId = taskLauncher().launch(request);
+		awaitWithPollAndTimeout(deploymentTimeout())
+				.untilAsserted(() -> assertThat(taskLauncher().status(launchId).getState()).isEqualTo(LaunchState.running));
 
+		List<Pod> pods = getPodsForTask(taskName);
 		assertThat(pods).hasSize(1);
+		assertThat(PodStatusUtil.isRunning(pods.get(0))).isTrue();
 
-		Pod pod = pods.get(0);
-
-		assertThat(pod.getSpec().getContainers()).hasSize(2);
-
-		Optional<Container> additionalContainer = pod.getSpec().getContainers().stream().filter(i -> i.getName().equals("test")).findFirst();
-		assertThat(additionalContainer.isPresent()).as("Additional container not found").isTrue();
-
-		Container testAdditionalContainer = additionalContainer.get();
-
-		assertThat(testAdditionalContainer.getName()).as("Unexpected additional container name").isEqualTo("test");
-		assertThat(testAdditionalContainer.getImage()).as("Unexpected additional container image").isEqualTo("busybox:latest");
-
-		List<String> commands = testAdditionalContainer.getCommand();
-
-		assertThat(commands).contains("sh", "-c", "echo hello");
-
-		log.info("Destroying {}...", taskName);
-
-		timeout = undeploymentTimeout();
-		kubernetesTaskLauncher.destroy(taskName);
+		log.info("Cleaning up {}...", taskName);
```

Comment:
What are your thoughts on having this on a ` finally` block ?   So we can attempt a cleanup if something in the test fails.

Reply:
👋🏻 @cppwfs ,

I definitely considered this case and should have left a comment. Here is why I left it w/o the finally:

```java 
TLW = TaskLauncherWrapper
```

This test is similiar to all of the others w/ the exception that it calls `TLW.cleanup()` rather than `TLW.destroy()`. The other tests rely on the parent's `@AfterEach` which handles any stragglers:
```java
@AfterEach
public void cleanupLingeringApps() {
  for (String id : launcherWrapper.launchedTasks) {
    try {
      launcherWrapper.wrapped.cleanup(id);
    }
    catch (Exception e) {
      log.warn("Exception caught while trying to cleanup '{}'. Moving on...", id);
    }
  }
  for (String appName : launcherWrapper.deployedApps) {
    try {
      log.warn("Test named '{}' left behind an app for ''. Trying to destroy.", this.testName, appName);
      launcherWrapper.wrapped.destroy(appName);
    }
    catch (Exception e) {
      log.warn("Exception caught while trying to destroy '{}'. Moving on...", appName);
    }
  }
}
```
I believe this will handle this new test case as well.

Reply:
Doh!   Thanks for the explanation,  I forgot to check the parent!